alchemist-web-renderer/src/jvmMain/kotlin/it/unibo/alchemist/server/routes/SimulationRoute.kt

```diff
@@ -0,0 +1,108 @@
+import it.unibo.alchemist.server.utility.Response.Companion.respond
+import it.unibo.alchemist.server.state.ServerStore.store
+import it.unibo.alchemist.server.surrogates.utility.toStatusSurrogate
+import it.unibo.alchemist.server.utility.Response
+
+/**
+ * Logic of the Routes in the /simulation path.
+ */
+object SimulationRoute {
+
+    /**
+     * Route that retrieve the simulation status and return it to the client mapping it with the [toStatusSurrogate]
+     * function.
+     * The HTTP [Response] sent to the client can be of type:
+     * 200 (OK) if the status is correctly retrieved;
+     * 500 (Server Error) if the Simulation was not loaded correctly.
+     */
+    fun Route.simulationStatus() {
+        get(simulationStatusPath) {
+            respond(
+                store.state.simulation?.status?.let { status ->
+                    Response(OK, status.toStatusSurrogate())
+                } ?: Response(InternalServerError, "Simulation not loaded.")
+            )
+        }
+    }
+
+    /**
+     * Route that plays the simulation by calling the [it.unibo.alchemist.core.interfaces.Simulation.play] method.
+     * The HTTP [Response] sent to the client can be of type:
+     * 200 (OK) if the operation succeed;
+     * 409 (Conflict) if the simulation is already running, or is in an invalid state to call the play operation;
+     * 500 (Server Error) if the Simulation was not loaded correctly.
+     */
+    fun Route.simulationActionPlay() {
+        post(simulationPlayPath) {
+            respondAction(
+                { simulation -> simulation.play() },
+                additionalCheck = Pair(Status.RUNNING, "The Simulation is already running.")
+            )
+        }
+    }
+
+    /**
+     * Route that pauses the simulation by calling the [it.unibo.alchemist.core.interfaces.Simulation.pause] method.
+     * The HTTP [Response] sent to the client can be of type:
+     * 200 (OK) if the operation succeed;
+     * 409 (Conflict) if the simulation is already paused, or is in an invalid state to call the pause operation;
+     * 500 (Server Error) if the Simulation was not loaded correctly.
+     */
+    fun Route.simulationActionPause() {
+        post(simulationPausePath) {
+            respondAction(
+                { simulation -> simulation.pause() },
+                additionalCheck = Pair(Status.PAUSED, "The Simulation is already paused.")
+            )
+        }
+    }
+
+    /**
+     * Private function that checks the simulation Status, executes the requested action if possible and responds.
+     * @param action the action to execute on the simulation.
+     * @param additionalCheck a [Pair] representing an additional Status check, first is an invalid [Status] and second
+     * is the error message.
+     */
+    private suspend fun PipelineContext<Unit, ApplicationCall>.respondAction(
+        action: (Simulation<Any, Nothing>) -> Unit,
+        additionalCheck: Pair<Status, String>
```

Comment:
```suggestion
        additionalCheck: Pair<Status, String>,
        action: (Simulation<Any, Nothing>) -> Unit,
```

Always make functions the last argument to enjoy the trailing lambda convention

Reply:
That is correct, will do